src/visualization/context/pagination.tsx

```diff
@@ -53,7 +53,9 @@ export const PaginationProvider: FC<PaginationProviderProps> = ({
 
   const setPage = useCallback(
     (page: number) => {
-      setOffset(Math.min(Math.max(0, (page - 1) * size), total - size))
+      setOffset(
+        Math.min(Math.max(0, (page - 1) * size), Math.max(0, total - size))
```

Comment:
Quick unit test!
example:
page: 1
size: 2
total: 11

Now see if offset is correct.

Reply:
good catch. This ended up being a pretty involved fix but take a look at what I ended up doing and let me know your thoughts.

Reply:
will approve as soon as you push up your fix again.